x-pack/plugins/reporting/public/share_context_menu/csv_export_modal.tsx

```diff
@@ -54,25 +54,20 @@ export const CsvModalContentUI: FC<Props> = (props: Props) => {
   const [absoluteUrl, setAbsoluteUrl] = useState('');
   const exceedsMaxLength = absoluteUrl.length >= getMaxUrlLength();
 
-  const getAbsoluteReportGenerationUrl = useCallback(() => {
-    const relativePath = apiClient.getReportingPublicJobPath(
-      'csv',
-      apiClient.getDecoratedJobParams(getJobParams(true) as unknown as AppParams)
-    );
-    return url.resolve(window.location.href, relativePath);
-  }, [apiClient, getJobParams]);
-
-  const setAbsoluteReportGenerationUrl = useCallback(() => {
-    if (!isMounted || !getAbsoluteReportGenerationUrl()) {
-      return;
-    } else {
-      setAbsoluteUrl(getAbsoluteReportGenerationUrl()!);
-    }
-  }, [isMounted, getAbsoluteReportGenerationUrl]);
+  const getAbsoluteReportGenerationUrl = useMemo(
+    () => () => {
+      const relativePath = apiClient.getReportingPublicJobPath(
+        CSV_JOB_TYPE,
+        apiClient.getDecoratedJobParams(getJobParams(true))
+      );
+      return setAbsoluteUrl(url.resolve(window.location.href, relativePath));
+    },
+    [apiClient, getJobParams]
+  );
 
   useEffect(() => {
-    setAbsoluteReportGenerationUrl();
-  }, [setAbsoluteReportGenerationUrl]);
+    getAbsoluteReportGenerationUrl();
+  }, [getAbsoluteReportGenerationUrl]);
```

Comment:
I think we could condense this like so;

```suggestion
 useEffect(() => {
   const reportingUrl = new URL(window.location.origin);
   reportingUrl.pathname = apiClient.getReportingPublicJobPath(
        CSV_JOB_TYPE,
        apiClient.getDecoratedJobParams(getJobParams(true))
      );
   setAbsoluteUrl(reportingUrl.toString())
 }, [apiClient, getJobParams])
```

and get the same result

Reply:
Ohohohoho nice!!!! I like it - tests are passing locally too, thanks!